packages/taquito-michel-codec/src/michelson-typecheck.ts

```diff
@@ -1847,10 +1851,22 @@ function functionTypeInternal(
       case 'TICKET': {
         const s = args(0, null, ['nat'])[0];
         ensureComparableType(s);
-        return [
-          annotate({ prim: 'ticket', args: [s] }, instructionAnn({ t: 1, v: 1 })),
-          ...stack.slice(2),
-        ];
+        if (ctx?.protocol === Protocol.PtLimaPtL) {
```

Comment:
Could you revert the if-else statement as follow:
```
if (ProtoInferiorTo(proto, Protocol.PtLimaPtL) {
         return [
            annotate({ prim: 'ticket', args: [s] }, instructionAnn({ t: 1, v: 1 })),
            ...stack.slice(2),
          ];
        } else {
return [
            annotateVar({
              prim: 'option',
              args: [
                annotate({ prim: 'ticket', args: [s] }, instructionAnn({ t: 1, v: 1 }))
              ]
            }),
            ...stack.slice(2)
          ];
        }

That way, we don't have to update for upcoming protocol, and everything will work with alpha (mondaynet)

Reply:
right! i saw that function but forgot to use it! thanks